legend-engine-language-pure-compiler/src/test/java/org/finos/legend/engine/language/pure/compiler/test/fromGrammar/TestDomainCompilationFromGrammar.java

```diff
@@ -2175,4 +2175,67 @@ public void testCompilationOfLambdaWithBiTemporalClass()
                 "    main::Person.all(%2020-12-12, %2020-12-13).firm(%latest, %latest)  \n" +
                 "} \n");
     }
+
+    @Test
+    public void testDatePropagationFromBusinessTemporalSourceToBusinessTemporalTarget()
+    {
+        test("###Pure\n" +
+                "Class <<temporal.businesstemporal>> main::Person\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "  firm: main::Firm[1];\n" +
+                "}\n" +
+                "\n" +
+                "Class <<temporal.businesstemporal>> main::Firm\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree(): String[*]\n" +
+                "{\n" +
+                "   main::Person.all(%2022-12-12).firm.name\n" +
+                "}\n");
+    }
+
+    @Test
+    public void testDatePropagationFromBiTemporalSourceToBusinessTemporalTarget()
+    {
+        test("###Pure\n" +
+                "Class <<temporal.bitemporal>> main::Person\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "  firm: main::Firm[1];\n" +
+                "}\n" +
+                "\n" +
+                "Class <<temporal.businesstemporal>> main::Firm\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree(): String[*]\n" +
+                "{\n" +
+                "   main::Person.all(%2022-12-12, %2022-12-12).firm.name\n" +
+                "}\n");
+    }
+
+    @Test
+    public void testDatePropagationNotSupportedFromProcessingTemporalSourceToBusinessTemporalTarget()
+    {
+        test("###Pure\n" +
+                "Class <<temporal.processingtemporal>> main::Person\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "  firm: main::Firm[1];\n" +
+                "}\n" +
+                "\n" +
+                "Class <<temporal.businesstemporal>> main::Firm\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree(): String[*]\n" +
+                "{\n" +
+                "   main::Person.all(%2022-12-12, %2022-12-12).firm.name\n" +
+                "}\n", "COMPILATION error at [13:1-16:1]: Error in 'main::walkTree': No-Arg milestoned property: 'firm' must be either called in a milestoning context or supplied with [businessDate] parameters");
+    }
```

Comment:
Can we add few more tests? - source is non-temporal, target is bitemporal (with both dates propagated), target is bitemporal (with one date propagated)

Also, we can do assertions on the built graph (to see whether propagation is happening as expected)